<?xml version="1.0"?>
<launch>
	<group ns="cargo_snap">
		<group ns="orient_strafing"> <!--TODO Tune these values on a full weight robot -->
	      <node name="orient_strafing_pid" pkg="pid" type="controller" >
	        <param name="node_name" value="orient_strafing_pid" />
	        <param name="Kp" value="3.5" />
	        <param name="Ki" value="0.0" />
	        <param name="Kd" value="0.015" />
	        <param name="upper_limit" value="7.5" />
	        <param name="lower_limit" value="-7.5" />
	        <param name="windup_limit" value="7.5" />
	        <param name="angle_error" value="true" />
	      </node>
		  <!-- TODO : orient strafind node requires values from here
				 fix it so that they can be read from the ones in the teleop namespace
		  -->
		  <rosparam file="$(find teleop_joystick_control)/config/teleop_joystick_comp_2022.yaml" command="load"/>
	      <node name="orient_strafing_node" pkg="teleop_joystick_control" type="orient_strafing_node" output="screen"/>
    </group>
    <node name="snap_to_cargo_2022_node" pkg="behaviors" type="2022_snap_to_cargo_node" output="screen"/>
  </group>
</launch>
